scaaml/capture/input_generators/attack_point_iterator_models.py

```diff
@@ -165,3 +166,177 @@ def items(self) -> AttackPointIteratorT:
     def get_generated_keys(self) -> List[str]:
         """Returns an exhaustive list of names this iterator will create."""
         return self.configuration.get_generated_keys()
+
+
+class ZipIteratorModel(BaseModel):
+    """
+    Attack point iterator zip pydantic model. This class takes any amount of
+    operands and combines them just like the `zip` function in Python.
+    
+    Args:
+        operation (Literal['zip']): The operation of the iterator
+            represents what the iterator does and what 
+            has to be in the config file. This is only used once to
+            double check if the operation is the correct one.
+            
+        operands (List[Union[BasicIteratorModels, RepeatIteratorModel]]):
+            The operands are any number of BasicIteratorModels or
+            RepeatIteratorModels that will be combined.
+    """
+    operation: Literal["zip"]
+    operands: List[Union[BasicIteratorModels, RepeatIteratorModel]]
+
+    def __len__(self) -> int:
+        non_negative_lengths: List[int] = []
+        for operand in self.operands:
+            try:
+                if len(operand) >= 0:
```

Comment:
This always holds.

Reply:
`len` always returns a non-negative number. That is why there is the `InfiniteLengthException`. So this if condition is not needed (just the line inside it would be doing exactly the same).